packages/core/contracts/financial-templates/common/financial-product-libraries/KpiOptionsFinancialProductLibrary.sol

```diff
@@ -11,6 +11,33 @@ import "../../../common/implementation/Lockable.sol";
  * Post-expiry, the collateral requirement is left as 1 and the price is left unchanged.
  */
 contract KpiOptionsFinancialProductLibrary is FinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    struct TransformedPrice {
+        bool set;
+        FixedPoint.Unsigned price;
+    }
+
+    mapping(address => TransformedPrice) public financialProductTransformedPrices;
```

Comment:
@eng is there any reason to make this mapping private (and also permission `transformPrice` to only be called by contracts) in consideration of oracle parasitic usage? I'm guessing its not a problem, but want to highlight

Reply:
I dont think that there is any parasitic usage on this. This is a transformation so it's not storing any price information that other contracts could use in any meaningful way. IMHO it's better to leave it public to ensure that we can call it to verify the state of set transformations.